pkg/plugins/trivy/image.go

```diff
@@ -588,84 +588,85 @@ func getCommandAndArgs(ctx trivyoperator.PluginContext, mode Mode, imageRef, tri
+	if sbomSources != "" {
+		args = append(args, []string{"--sbom-sources", sbomSources}...)
 	}
-	imcs := imageConfigSecretScanner(trivyConfig)
-	var imageconfigSecretScannerFlag string
-	if len(imcs) == 2 {
-		imageconfigSecretScannerFlag = fmt.Sprintf("%s %s ", imcs[0], imcs[1])
+
+	scanners := Scanners(trivyConfig)
+	args = append(args, scanners, getSecurityChecks(ctx))
+
+	if trivyServerURL != "" {
+		args = append(args, []string{"--server", trivyServerURL}...)
 	}
+
 	var skipUpdate string
-	if c.GetClientServerSkipUpdate() && mode == ClientServer {
-		skipUpdate = SkipDBUpdate(c)
+	if trivyConfig.GetClientServerSkipUpdate() && mode == ClientServer {
+		skipUpdate = SkipDBUpdate(trivyConfig)
 	} else if mode != ClientServer {
-		skipUpdate = SkipDBUpdate(c)
+		skipUpdate = SkipDBUpdate(trivyConfig)
+	}
+	if skipUpdate != "" {
+		args = append(args, skipUpdate)
 	}
-	if !compressLogs {
-		args := []string{
-			"--cache-dir",
-			cacheDir,
-			"--quiet",
-			"image",
-			scanners,
-			getSecurityChecks(ctx),
-			"--format",
-			"json",
-		}
-		if trivyServerURL != "" {
-			args = append(args, []string{"--server", trivyServerURL}...)
-		}
-		args = append(args, imageRef)
 
-		if slow != "" {
-			args = append(args, slow)
-		}
-		if len(vulnTypeArgs) > 0 {
-			args = append(args, vulnTypeArgs...)
-		}
-		if len(imcs) > 0 {
-			args = append(args, imcs...)
-		}
-		pkgList := getPkgList(ctx)
-		if pkgList != "" {
-			args = append(args, pkgList)
-		}
-		if sbomSources != "" {
-			args = append(args, []string{"--sbom-sources", sbomSources}...)
-		}
-		if skipUpdate != "" {
-			args = append(args, skipUpdate)
-		}
-		if skipJavaDBUpdate != "" {
-			args = append(args, skipJavaDBUpdate)
-		}
+	skipJavaDBUpdate := SkipJavaDBUpdate(trivyConfig)
+	if skipJavaDBUpdate != "" {
+		args = append(args, skipJavaDBUpdate)
+	}
 
-		return command, args
+	slow := Slow(trivyConfig)
+	if slow != "" {
+		args = append(args, slow)
 	}
-	var serverUrlParms string
-	if mode == ClientServer {
-		serverUrlParms = fmt.Sprintf("--server '%s' ", trivyServerURL)
+
+	vulnTypeArgs := vulnTypeFilter(ctx)
+	if len(vulnTypeArgs) > 0 {
+		args = append(args, vulnTypeArgs...)
 	}
-	var sbomSourcesFlag string
-	if sbomSources != "" {
-		sbomSourcesFlag = fmt.Sprintf(" --sbom-sources %s ", sbomSources)
+
+	pkgList := getPkgList(ctx)
+	if pkgList != "" {
+		args = append(args, pkgList)
 	}
-	return []string{"/bin/sh"}, []string{"-c", fmt.Sprintf(`trivy image %s '%s' %s %s %s %s %s %s%s --cache-dir %s --quiet %s --format json %s> /tmp/scan/%s &&  bzip2 -c /tmp/scan/%s | base64`, slow, imageRef, scanners, getSecurityChecks(ctx), imageconfigSecretScannerFlag, vulnTypeFlag, skipUpdate, skipJavaDBUpdate, sbomSourcesFlag, cacheDir, getPkgList(ctx), serverUrlParms, resultFileName, resultFileName)}
+
+	// Return early when compressing logs is disabled.
+	compressLogs := trivyOperatorConfig.CompressLogs()
+	if !compressLogs {
+		return command, args
```

Comment:
```suggestion
		return []string{"trivy"}, args
```

Reply:
there is only one using of `command`

Reply:
There is another usage on line 664. If you still want `[]string{"trivy"}` instead I'm fine changing it though!